


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: CertPathValidatorException</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">java.security.cert</a> ]
</div>

<h1>Coverage Summary for Class: CertPathValidatorException (java.security.cert)</h1>

<table class="coverageStats">

<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">CertPathValidatorException</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 10)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 29)
  </span>
</td>
</tr>
  <tr>
    <td class="name">CertPathValidatorException$BasicReason</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 8)
  </span>
</td>
  </tr>
<tr>
  <td class="name"><strong>total</strong></td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 11)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 37)
  </span>
</td>
</tr>
</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;/*
<i>2</i>&nbsp; * Copyright (c) 2000, 2013, Oracle and/or its affiliates. All rights reserved.
<i>3</i>&nbsp; * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
<i>4</i>&nbsp; *
<i>5</i>&nbsp; *
<i>6</i>&nbsp; *
<i>7</i>&nbsp; *
<i>8</i>&nbsp; *
<i>9</i>&nbsp; *
<i>10</i>&nbsp; *
<i>11</i>&nbsp; *
<i>12</i>&nbsp; *
<i>13</i>&nbsp; *
<i>14</i>&nbsp; *
<i>15</i>&nbsp; *
<i>16</i>&nbsp; *
<i>17</i>&nbsp; *
<i>18</i>&nbsp; *
<i>19</i>&nbsp; *
<i>20</i>&nbsp; *
<i>21</i>&nbsp; *
<i>22</i>&nbsp; *
<i>23</i>&nbsp; *
<i>24</i>&nbsp; */
<i>25</i>&nbsp;
<i>26</i>&nbsp;package java.security.cert;
<i>27</i>&nbsp;
<i>28</i>&nbsp;import java.io.InvalidObjectException;
<i>29</i>&nbsp;import java.io.IOException;
<i>30</i>&nbsp;import java.io.ObjectInputStream;
<i>31</i>&nbsp;import java.security.GeneralSecurityException;
<i>32</i>&nbsp;
<i>33</i>&nbsp;/**
<i>34</i>&nbsp; * An exception indicating one of a variety of problems encountered when
<i>35</i>&nbsp; * validating a certification path.
<i>36</i>&nbsp; * &lt;p&gt;
<i>37</i>&nbsp; * A {@code CertPathValidatorException} provides support for wrapping
<i>38</i>&nbsp; * exceptions. The {@link #getCause getCause} method returns the throwable,
<i>39</i>&nbsp; * if any, that caused this exception to be thrown.
<i>40</i>&nbsp; * &lt;p&gt;
<i>41</i>&nbsp; * A {@code CertPathValidatorException} may also include the
<i>42</i>&nbsp; * certification path that was being validated when the exception was thrown,
<i>43</i>&nbsp; * the index of the certificate in the certification path that caused the
<i>44</i>&nbsp; * exception to be thrown, and the reason that caused the failure. Use the
<i>45</i>&nbsp; * {@link #getCertPath getCertPath}, {@link #getIndex getIndex}, and
<i>46</i>&nbsp; * {@link #getReason getReason} methods to retrieve this information.
<i>47</i>&nbsp; *
<i>48</i>&nbsp; * &lt;p&gt;
<i>49</i>&nbsp; * &lt;b&gt;Concurrent Access&lt;/b&gt;
<i>50</i>&nbsp; * &lt;p&gt;
<i>51</i>&nbsp; * Unless otherwise specified, the methods defined in this class are not
<i>52</i>&nbsp; * thread-safe. Multiple threads that need to access a single
<i>53</i>&nbsp; * object concurrently should synchronize amongst themselves and
<i>54</i>&nbsp; * provide the necessary locking. Multiple threads each manipulating
<i>55</i>&nbsp; * separate objects need not synchronize.
<i>56</i>&nbsp; *
<i>57</i>&nbsp; * @see CertPathValidator
<i>58</i>&nbsp; *
<i>59</i>&nbsp; * @since       1.4
<i>60</i>&nbsp; * @author      Yassir Elley
<i>61</i>&nbsp; */
<i>62</i>&nbsp;public class CertPathValidatorException extends GeneralSecurityException {
<i>63</i>&nbsp;
<i>64</i>&nbsp;    private static final long serialVersionUID = -3083180014971893139L;
<i>65</i>&nbsp;
<i>66</i>&nbsp;    /**
<i>67</i>&nbsp;     * @serial the index of the certificate in the certification path
<i>68</i>&nbsp;     * that caused the exception to be thrown
<i>69</i>&nbsp;     */
<b class="nc"><i>70</i>&nbsp;    private int index = -1;</b>
<i>71</i>&nbsp;
<i>72</i>&nbsp;    /**
<i>73</i>&nbsp;     * @serial the {@code CertPath} that was being validated when
<i>74</i>&nbsp;     * the exception was thrown
<i>75</i>&nbsp;     */
<i>76</i>&nbsp;    private CertPath certPath;
<i>77</i>&nbsp;
<i>78</i>&nbsp;    /**
<i>79</i>&nbsp;     * @serial the reason the validation failed
<i>80</i>&nbsp;     */
<b class="nc"><i>81</i>&nbsp;    private Reason reason = BasicReason.UNSPECIFIED;</b>
<i>82</i>&nbsp;
<i>83</i>&nbsp;    /**
<i>84</i>&nbsp;     * Creates a {@code CertPathValidatorException} with
<i>85</i>&nbsp;     * no detail message.
<i>86</i>&nbsp;     */
<i>87</i>&nbsp;    public CertPathValidatorException() {
<b class="nc"><i>88</i>&nbsp;        this(null, null);</b>
<i>89</i>&nbsp;    }
<i>90</i>&nbsp;
<i>91</i>&nbsp;    /**
<i>92</i>&nbsp;     * Creates a {@code CertPathValidatorException} with the given
<i>93</i>&nbsp;     * detail message. A detail message is a {@code String} that
<i>94</i>&nbsp;     * describes this particular exception.
<i>95</i>&nbsp;     *
<i>96</i>&nbsp;     * @param msg the detail message
<i>97</i>&nbsp;     */
<i>98</i>&nbsp;    public CertPathValidatorException(String msg) {
<b class="nc"><i>99</i>&nbsp;        this(msg, null);</b>
<i>100</i>&nbsp;    }
<i>101</i>&nbsp;
<i>102</i>&nbsp;    /**
<i>103</i>&nbsp;     * Creates a {@code CertPathValidatorException} that wraps the
<i>104</i>&nbsp;     * specified throwable. This allows any exception to be converted into a
<i>105</i>&nbsp;     * {@code CertPathValidatorException}, while retaining information
<i>106</i>&nbsp;     * about the wrapped exception, which may be useful for debugging. The
<i>107</i>&nbsp;     * detail message is set to ({@code cause==null ? null : cause.toString()})
<i>108</i>&nbsp;     * (which typically contains the class and detail message of
<i>109</i>&nbsp;     * cause).
<i>110</i>&nbsp;     *
<i>111</i>&nbsp;     * @param cause the cause (which is saved for later retrieval by the
<i>112</i>&nbsp;     * {@link #getCause getCause()} method). (A {@code null} value is
<i>113</i>&nbsp;     * permitted, and indicates that the cause is nonexistent or unknown.)
<i>114</i>&nbsp;     */
<i>115</i>&nbsp;    public CertPathValidatorException(Throwable cause) {
<b class="nc"><i>116</i>&nbsp;        this((cause == null ? null : cause.toString()), cause);</b>
<i>117</i>&nbsp;    }
<i>118</i>&nbsp;
<i>119</i>&nbsp;    /**
<i>120</i>&nbsp;     * Creates a {@code CertPathValidatorException} with the specified
<i>121</i>&nbsp;     * detail message and cause.
<i>122</i>&nbsp;     *
<i>123</i>&nbsp;     * @param msg the detail message
<i>124</i>&nbsp;     * @param cause the cause (which is saved for later retrieval by the
<i>125</i>&nbsp;     * {@link #getCause getCause()} method). (A {@code null} value is
<i>126</i>&nbsp;     * permitted, and indicates that the cause is nonexistent or unknown.)
<i>127</i>&nbsp;     */
<i>128</i>&nbsp;    public CertPathValidatorException(String msg, Throwable cause) {
<b class="nc"><i>129</i>&nbsp;        this(msg, cause, null, -1);</b>
<i>130</i>&nbsp;    }
<i>131</i>&nbsp;
<i>132</i>&nbsp;    /**
<i>133</i>&nbsp;     * Creates a {@code CertPathValidatorException} with the specified
<i>134</i>&nbsp;     * detail message, cause, certification path, and index.
<i>135</i>&nbsp;     *
<i>136</i>&nbsp;     * @param msg the detail message (or {@code null} if none)
<i>137</i>&nbsp;     * @param cause the cause (or {@code null} if none)
<i>138</i>&nbsp;     * @param certPath the certification path that was in the process of
<i>139</i>&nbsp;     * being validated when the error was encountered
<i>140</i>&nbsp;     * @param index the index of the certificate in the certification path
<i>141</i>&nbsp;     * that caused the error (or -1 if not applicable). Note that
<i>142</i>&nbsp;     * the list of certificates in a {@code CertPath} is zero based.
<i>143</i>&nbsp;     * @throws IndexOutOfBoundsException if the index is out of range
<i>144</i>&nbsp;     * {@code (index &lt; -1 || (certPath != null &amp;&amp; index &gt;=
<i>145</i>&nbsp;     * certPath.getCertificates().size()) }
<i>146</i>&nbsp;     * @throws IllegalArgumentException if {@code certPath} is
<i>147</i>&nbsp;     * {@code null} and {@code index} is not -1
<i>148</i>&nbsp;     */
<i>149</i>&nbsp;    public CertPathValidatorException(String msg, Throwable cause,
<i>150</i>&nbsp;            CertPath certPath, int index) {
<b class="nc"><i>151</i>&nbsp;        this(msg, cause, certPath, index, BasicReason.UNSPECIFIED);</b>
<i>152</i>&nbsp;    }
<i>153</i>&nbsp;
<i>154</i>&nbsp;    /**
<i>155</i>&nbsp;     * Creates a {@code CertPathValidatorException} with the specified
<i>156</i>&nbsp;     * detail message, cause, certification path, index, and reason.
<i>157</i>&nbsp;     *
<i>158</i>&nbsp;     * @param msg the detail message (or {@code null} if none)
<i>159</i>&nbsp;     * @param cause the cause (or {@code null} if none)
<i>160</i>&nbsp;     * @param certPath the certification path that was in the process of
<i>161</i>&nbsp;     * being validated when the error was encountered
<i>162</i>&nbsp;     * @param index the index of the certificate in the certification path
<i>163</i>&nbsp;     * that caused the error (or -1 if not applicable). Note that
<i>164</i>&nbsp;     * the list of certificates in a {@code CertPath} is zero based.
<i>165</i>&nbsp;     * @param reason the reason the validation failed
<i>166</i>&nbsp;     * @throws IndexOutOfBoundsException if the index is out of range
<i>167</i>&nbsp;     * {@code (index &lt; -1 || (certPath != null &amp;&amp; index &gt;=
<i>168</i>&nbsp;     * certPath.getCertificates().size()) }
<i>169</i>&nbsp;     * @throws IllegalArgumentException if {@code certPath} is
<i>170</i>&nbsp;     * {@code null} and {@code index} is not -1
<i>171</i>&nbsp;     * @throws NullPointerException if {@code reason} is {@code null}
<i>172</i>&nbsp;     *
<i>173</i>&nbsp;     * @since 1.7
<i>174</i>&nbsp;     */
<i>175</i>&nbsp;    public CertPathValidatorException(String msg, Throwable cause,
<i>176</i>&nbsp;            CertPath certPath, int index, Reason reason) {
<b class="nc"><i>177</i>&nbsp;        super(msg, cause);</b>
<b class="nc"><i>178</i>&nbsp;        if (certPath == null &amp;&amp; index != -1) {</b>
<b class="nc"><i>179</i>&nbsp;            throw new IllegalArgumentException();</b>
<i>180</i>&nbsp;        }
<b class="nc"><i>181</i>&nbsp;        if (index &lt; -1 ||</b>
<b class="nc"><i>182</i>&nbsp;            (certPath != null &amp;&amp; index &gt;= certPath.getCertificates().size())) {</b>
<b class="nc"><i>183</i>&nbsp;            throw new IndexOutOfBoundsException();</b>
<i>184</i>&nbsp;        }
<b class="nc"><i>185</i>&nbsp;        if (reason == null) {</b>
<b class="nc"><i>186</i>&nbsp;            throw new NullPointerException(&quot;reason can&#39;t be null&quot;);</b>
<i>187</i>&nbsp;        }
<b class="nc"><i>188</i>&nbsp;        this.certPath = certPath;</b>
<b class="nc"><i>189</i>&nbsp;        this.index = index;</b>
<b class="nc"><i>190</i>&nbsp;        this.reason = reason;</b>
<i>191</i>&nbsp;    }
<i>192</i>&nbsp;
<i>193</i>&nbsp;    /**
<i>194</i>&nbsp;     * Returns the certification path that was being validated when
<i>195</i>&nbsp;     * the exception was thrown.
<i>196</i>&nbsp;     *
<i>197</i>&nbsp;     * @return the {@code CertPath} that was being validated when
<i>198</i>&nbsp;     * the exception was thrown (or {@code null} if not specified)
<i>199</i>&nbsp;     */
<i>200</i>&nbsp;    public CertPath getCertPath() {
<b class="nc"><i>201</i>&nbsp;        return this.certPath;</b>
<i>202</i>&nbsp;    }
<i>203</i>&nbsp;
<i>204</i>&nbsp;    /**
<i>205</i>&nbsp;     * Returns the index of the certificate in the certification path
<i>206</i>&nbsp;     * that caused the exception to be thrown. Note that the list of
<i>207</i>&nbsp;     * certificates in a {@code CertPath} is zero based. If no
<i>208</i>&nbsp;     * index has been set, -1 is returned.
<i>209</i>&nbsp;     *
<i>210</i>&nbsp;     * @return the index that has been set, or -1 if none has been set
<i>211</i>&nbsp;     */
<i>212</i>&nbsp;    public int getIndex() {
<b class="nc"><i>213</i>&nbsp;        return this.index;</b>
<i>214</i>&nbsp;    }
<i>215</i>&nbsp;
<i>216</i>&nbsp;    /**
<i>217</i>&nbsp;     * Returns the reason that the validation failed. The reason is
<i>218</i>&nbsp;     * associated with the index of the certificate returned by
<i>219</i>&nbsp;     * {@link #getIndex}.
<i>220</i>&nbsp;     *
<i>221</i>&nbsp;     * @return the reason that the validation failed, or
<i>222</i>&nbsp;     *    {@code BasicReason.UNSPECIFIED} if a reason has not been
<i>223</i>&nbsp;     *    specified
<i>224</i>&nbsp;     *
<i>225</i>&nbsp;     * @since 1.7
<i>226</i>&nbsp;     */
<i>227</i>&nbsp;    public Reason getReason() {
<b class="nc"><i>228</i>&nbsp;        return this.reason;</b>
<i>229</i>&nbsp;    }
<i>230</i>&nbsp;
<i>231</i>&nbsp;    private void readObject(ObjectInputStream stream)
<i>232</i>&nbsp;        throws ClassNotFoundException, IOException {
<b class="nc"><i>233</i>&nbsp;        stream.defaultReadObject();</b>
<b class="nc"><i>234</i>&nbsp;        if (reason == null) {</b>
<b class="nc"><i>235</i>&nbsp;            reason = BasicReason.UNSPECIFIED;</b>
<i>236</i>&nbsp;        }
<b class="nc"><i>237</i>&nbsp;        if (certPath == null &amp;&amp; index != -1) {</b>
<b class="nc"><i>238</i>&nbsp;            throw new InvalidObjectException(&quot;certpath is null and index != -1&quot;);</b>
<i>239</i>&nbsp;        }
<b class="nc"><i>240</i>&nbsp;        if (index &lt; -1 ||</b>
<b class="nc"><i>241</i>&nbsp;            (certPath != null &amp;&amp; index &gt;= certPath.getCertificates().size())) {</b>
<b class="nc"><i>242</i>&nbsp;            throw new InvalidObjectException(&quot;index out of range&quot;);</b>
<i>243</i>&nbsp;        }
<i>244</i>&nbsp;    }
<i>245</i>&nbsp;
<i>246</i>&nbsp;    /**
<i>247</i>&nbsp;     * The reason the validation algorithm failed.
<i>248</i>&nbsp;     *
<i>249</i>&nbsp;     * @since 1.7
<i>250</i>&nbsp;     */
<i>251</i>&nbsp;    public static interface Reason extends java.io.Serializable { }
<i>252</i>&nbsp;
<i>253</i>&nbsp;
<i>254</i>&nbsp;    /**
<i>255</i>&nbsp;     * The BasicReason enumerates the potential reasons that a certification
<i>256</i>&nbsp;     * path of any type may be invalid.
<i>257</i>&nbsp;     *
<i>258</i>&nbsp;     * @since 1.7
<i>259</i>&nbsp;     */
<b class="nc"><i>260</i>&nbsp;    public static enum BasicReason implements Reason {</b>
<i>261</i>&nbsp;        /**
<i>262</i>&nbsp;         * Unspecified reason.
<i>263</i>&nbsp;         */
<b class="nc"><i>264</i>&nbsp;        UNSPECIFIED,</b>
<i>265</i>&nbsp;
<i>266</i>&nbsp;        /**
<i>267</i>&nbsp;         * The certificate is expired.
<i>268</i>&nbsp;         */
<b class="nc"><i>269</i>&nbsp;        EXPIRED,</b>
<i>270</i>&nbsp;
<i>271</i>&nbsp;        /**
<i>272</i>&nbsp;         * The certificate is not yet valid.
<i>273</i>&nbsp;         */
<b class="nc"><i>274</i>&nbsp;        NOT_YET_VALID,</b>
<i>275</i>&nbsp;
<i>276</i>&nbsp;        /**
<i>277</i>&nbsp;         * The certificate is revoked.
<i>278</i>&nbsp;         */
<b class="nc"><i>279</i>&nbsp;        REVOKED,</b>
<i>280</i>&nbsp;
<i>281</i>&nbsp;        /**
<i>282</i>&nbsp;         * The revocation status of the certificate could not be determined.
<i>283</i>&nbsp;         */
<b class="nc"><i>284</i>&nbsp;        UNDETERMINED_REVOCATION_STATUS,</b>
<i>285</i>&nbsp;
<i>286</i>&nbsp;        /**
<i>287</i>&nbsp;         * The signature is invalid.
<i>288</i>&nbsp;         */
<b class="nc"><i>289</i>&nbsp;        INVALID_SIGNATURE,</b>
<i>290</i>&nbsp;
<i>291</i>&nbsp;        /**
<i>292</i>&nbsp;         * The public key or the signature algorithm has been constrained.
<i>293</i>&nbsp;         */
<b class="nc"><i>294</i>&nbsp;        ALGORITHM_CONSTRAINED</b>
<i>295</i>&nbsp;    }
<i>296</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2018-11-19 23:14</div>
</div>
</body>
</html>
